### PR TITLE
Configuration Variable Redesign

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

Redesign of the Configuration Variable for Integrations, Lovelace cards, etc

image

image

## Type of change

- [x] Style changes

## Additional information

- Linked to: #13207

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by frenck** [comment]
Nice! That is a good leap forward!

image


I do think the font is out of wack compared to the rest of the website...

The last line break should not be there I think, it is probably the cause of the additional margin at the bottom as well?

Lastly, what annoys me (in both this and the old version) is the default value. It feels so... dunno; dumped at the end?

Just an idea (which personally isn't fully satisfying my inner soul either):

image

**Comment by zsarnett** [comment]
The problem is that a default value could be a sentence rather than just a value. In the case of the weather card. I do like that idea though. 

The font is the same. Different size and the opacity is turned down. I feel like it separates it good enough but not too much. I also feel like the rest of the site should be different. But I'm working on that 😉

I'll take a look at the last line break. It's already something I do in the nested so easy to remove from the main

**Comment by frenck** [comment]
> The problem is that a default value could be a sentence rather than just a value. In the case of the weather card. I do like that idea though.

Well, since we are in a loop, we can determine the length of the default value and behave differently?
I'm kinda sure that at least 95% of the cases it contains a single word/number.

e.g., in case the default value is less than 20 chars, show it on top, else in the bottom.

**Comment by zsarnett** [comment]
> The last line break should not be there I think, it is probably the cause of the additional margin at the bottom as well?

This margin is from the h2 in the documentation for Tradfi

**Comment by zsarnett** [comment]
Also on the font. 

Would we be so crazy to change our main font presets from

```css
color: #222;
font-size: 18px;
```

to 

```css
color: #4f566b;
font-size: 16px;
```

### from: 
image


### to:
image


I find it incredibly unnecessary to have such large text as our base. and having the headers so large and bold and uppercased. I think we can do better and look more professional

**Comment by zsarnett** [comment]
image

File: plugins/configuration.rb [hunk 1]
```diff
@@ -145,7 +166,7 @@ def render(context)
 
       <<~MARKUP
         <div class="config-vars">
-          <h3><a class="title-link" name="configuration-variables" href="#configuration-variables"></a> Configuration Variables</h3>
+          <h4><a class="title-link" name="configuration-variables" href="#configuration-variables"></a> Configuration Variables</h4>
```
Comment (frenck) [inline plugins/configuration.rb]: This is always a H3, as the configuration heading is always H2. For SEO structure it is important to have that in order.

File: plugins/configuration.rb [hunk 2]
```diff
@@ -102,14 +127,10 @@ def render_config_vars(vars:, component:, platform:, converter:, classes: nil, p
...
-          markup << "<div class='config-vars-default'>\nDefault value: #{converter.convert(attr['default'].to_s)}</div>"
-        elsif attr['type'].to_s.include? 'boolean'
-          # Is the type is a boolean, a default key is required
-          raise ArgumentError, "Configuration key '#{key}' is a boolean type and"\
-            " therefore, requires a default."
+        if isDefault && defaultValue.length > MIN_DEFAULT_LENGTH
+          markup << "<div class='config-vars-default'>\nDefault: #{defaultValue}</div>"
         end
-
+
```
Comment (frenck) [inline plugins/configuration.rb]:
```suggestion

```

File: plugins/configuration.rb [hunk 3]
```diff
@@ -83,14 +85,37 @@ def render_config_vars(vars:, component:, platform:, converter:, classes: nil, p
             unless ['list', 'template'].include? parent_type
         end
 
+        defaultValue = ""
+        isDefault = false
+        if attr.key? 'default' and not attr['default'].to_s.empty? 
+          isDefault = true
+          defaultValue = converter.convert(attr['default'].to_s)
+        elsif attr['type'].to_s.include? 'boolean'
+          # Is the type is a boolean, a default key is required
```
Comment (frenck) [inline plugins/configuration.rb]: Not your fault, but easy fix:

```suggestion
          # If the type is a boolean, a default key is required
```

**Comment by frenck** [comment]
I agree that we can reduce the font size in general. It is huge!

Decreasing the contrast with the background by smoothing down the font color... Not sure if that is ideal for everybody. Mainly because of our dirty background (which I personally hate, looks like it was washed a thousand times).

If we can fix the h3/h4 and merge those 2 little other tiny things, then this should be ready to rock 🤘

**Comment by frenck** [comment]
Some older screenshots of experiments, those are actually live embedded demos

image


image

**Comment by frenck** [comment]
Some dark play and font test

image

image

**Comment by frenck** [comment]
All unrelated, and all have too many things going on / not right. Was just some idea gathering from the past.

So sorry about that 😉

File: plugins/configuration.rb (hunk 1 above)
Comment (zsarnett) [reply plugins/configuration.rb]: We should decrease the size and boldness of each heading eventually

File: plugins/configuration.rb [hunk 4]
```diff
@@ -149,7 +165,7 @@ def render(context)
 
       <<~MARKUP
         <div class="config-vars">
-          <h3><a class="title-link" name="configuration-variables" href="#configuration-variables"></a> Configuration Variables</h3>
+          <h4><a class="title-link" name="configuration-variables" href="#configuration-variables"></a> Configuration Variables</h4>
```
Comment (zsarnett) [inline plugins/configuration.rb]:
```suggestion
          <h3><a class="title-link" name="configuration-variables" href="#configuration-variables"></a> Configuration Variables</h3>
```

**Pull request merged**